packages/use-presence/src/index.tsx

```diff
@@ -107,3 +109,57 @@ export default function usePresence(
     isExiting
   };
 }
+
+export type UseUniqueDataPresenceOptions<T> = {
+  /** Check if two instances of type T are equal, defaults to a === comparison */
+  equalityCheck?(a: T, b: T): boolean;
+  /** Check if an instance of type T is considered valid to render, defaults to Boolean function */
+  validationCheck?(a: T): boolean;
+} & UsePresenceOptions;
+
+function defaultEqualityCheck<T>(a: T, b: T) {
+  return a === b;
+}
+function defaultValidationCheck<T>(a: T) {
+  return Boolean(a);
+}
+
+export function useUniqueDataPresence<T>(
+  dataInput: T,
+  {
+    equalityCheck = defaultEqualityCheck,
```

Comment:
Do we need to make this configurable? If we just assume reference equality, I think the consumer could implement a special equality check outside of the lib and achieve the same (e.g. with `useMemo`).

For libraries I'm largely in favour of avoiding options, especially if they can be solved outside. There's always a use case popping up at some point that an API didn't consider and if it can be achieved without changing the lib, I think that's useful 🙂.

Reply:
I don't need it to be configurable in any of my use-cases. I also agree that this logic can be moved outside of the library where it's needed. But reference checks meet the needs of all my scenarios.